airflow/www/views.py

```diff
@@ -1374,90 +1376,113 @@ def tree(self):
                 .all()
             )
         dag_runs = {
-            dr.execution_date: alchemy_to_dict(dr) for dr in dag_runs}
+            dr.execution_date: alchemy_to_dict(dr) for dr in dag_runs
+        }
 
         dates = sorted(list(dag_runs.keys()))
         max_date = max(dates) if dates else None
         min_date = min(dates) if dates else None
 
         tis = dag.get_task_instances(start_date=min_date, end_date=base_date)
-        task_instances = {}
+        task_instances: Dict[Tuple[str, datetime], Optional[models.TaskInstance]] = {}
```

Comment:
```suggestion
        task_instances: Dict[Tuple[str, datetime], models.TaskInstance] = {}
```

I think? i.e. we don't ever store None in the dict do we?